src/pc/1.9/chunk.js

```diff
@@ -0,0 +1,479 @@
+    return Buffer.concat([blocks, blocklights, skylights, biomes]);
+  }
+
+  readSection(section) {
+    try {
+      return Chunk.packingProtocol.read(section, 0, 'section', {});
+    } catch (e) {
+      e.message = `Read error for ${e.field} : ${e.message}`;
+      throw e;
+    }
+  }
+
+
+  //Simplified eatPackedBlockLongs Algorithm
+  eatPackedBlockLongs(rawBuffer, palette, bitsPerBlock) {
+    //The critical problem is that the internal order of each long is opposite to the organizational order of the longs
+    //This is easily fixed by flipping the order of the longs.
+    //Therefore, we will read 4 bytes at a time, bit-flip them, and write them into a new buffer.
+    //From there, the old algorithm for reading will work just fine, we don't even have to consider the existence of the longs anymore.
+    //A major side-effect, though, is that all of the internal block-datas will be flipped, so we have to flip them again before extracting data.
+    let unjumbledBuffer = Buffer.alloc(rawBuffer.length);
+    for (let l = 0; l < rawBuffer.length; l += 8) {
+      //Load the long
+      let longleftjumbled = rawBuffer.readUInt32BE(l, true);
+      let longrightjumbled = rawBuffer.readUInt32BE(l + 4, true);
+      //Write in reverse order -- flip bits by using little endian.
+      unjumbledBuffer.writeUInt32LE(longrightjumbled, l);
+      unjumbledBuffer.writeUInt32LE(longleftjumbled, l + 4);
+    }
+
+
+    let blockCount = unjumbledBuffer.length * 8 / bitsPerBlock;
+    let resultantBuffer = Buffer.alloc(blockCount * 2)
+    let localBit = 0;
+
+    for (let block = 0; block < blockCount; block++) {
+      //Determine the start-bit for the block.
+      let bit = block * bitsPerBlock;
+      //Determine the start-byte for that bit.
+      let targetbyte = Math.floor(bit / 8);
+
+      //Read a 32-bit section surrounding the targeted block
+
+      let datatarget = unjumbledBuffer.readUInt32LE(targetbyte, true);
+
+      //Determine the start bit local to the datatarget.
+      let localbit = bit % 8;
+
+      //Chop off uninteresting bits, then shift to that start bit:
+      let paletteid = (datatarget << (32 - localbit - bitsPerBlock)) >>> (32 - bitsPerBlock);
+
+      //Grab the data from the palette
+      let palettedata = paletteid;
+      if (palette.length != 0)
+        palettedata = palette[paletteid];
+      let data = palettedata & 0b1111;
+      let id = palettedata >>> 4;
+      resultantBuffer.writeUInt16LE((id << 4) | data, block * 2);
+    }
+    return resultantBuffer;
+  }
+
+
+  //This function turned out to be useless -- I just changed endianness on my reads/writes in clever spots
+  reverseBits(data, n) {
+    let storage = 0;
+    for (let i = 0; i < n; i++) {
+      storage = storage | (data & 1);
+      if (i != n - 1) {
+        storage = storage << 1;
+        data = data >> 1;
+      }
+    }
+    return storage;
+  }
+
+
+  /*
+  eatPackedBlockLongs(rawBuffer, palette, bitsPerBlock) {
```

Comment:
can this commented code be deleted ? (we can keep the commit if there's useful info in that)